src/Symfony/Bundle/FrameworkBundle/Test/KernelTestCase.php

```diff
@@ -165,10 +213,35 @@ protected static function createKernel(array $options = array())
             static::$class = static::getKernelClass();
         }
 
-        return new static::$class(
+        if (isset($_SERVER['KERNEL_ENV'])) {
+            $options['environment'] = $_SERVER['KERNEL_ENV'];
```

Comment:
options passed explicitly as argument must win over env variables IMO, otherwise this becomes a WTF for people passing options explicitly

Reply:
Replaced with current logic from 3.4 branch